cabal-install-solver/src/Distribution/Solver/Types/InstalledPreference.hs

```diff
@@ -7,5 +7,5 @@ import Prelude (Show)
 -- | Whether we prefer an installed version of a package or simply the latest
 -- version.
 --
-data InstalledPreference = PreferInstalled | PreferLatest
+data InstalledPreference = PreferInstalled | PreferLatest | PreferOldest
```

Comment:
Now as `cabal-install-solver` is a library, changes like this are **BREAKING CHNAGES**, so you don't simply backport them anymore to `cabal-install` releases.

I has warned about this problem, please deal with it appropriately. This change cannot be backported without a major version bump.

Reply:
I seem to remember `cabal-install-solver` designated as an "internal library", at least for the time being while API is shaking up. However, I don't see it described like that on Hackage, which perhaps should be fixed? I think either one or the other should be fixed.

Reply:
iirc in the end it was decided to honor pvp, and that's why there isn't any notice. i opened #8386 to revert the backport

Reply:
@Bodigrim: if 3.8.2.0 gets released and this PR is not included, please accept my apologies. I missed the API change and negotiated the inclusion believing the choice is 3.8.1.0 vs 3.8.2.0, not 3.8.1.0 vs 3.10.1.0. Perhaps we should just skip 3.8.2.0 regardless of anything and release 3.10.1.0 next, even if this PR is the only major change in it (but there is a lot of other changes accumulated on master already; I don't remember how breaking they are and it's to be seen if we'd base 3.10 off master or off current 3.8 with your API breaking change).